### ISSUE TITLE
open -[...] conflicts with negation

### ISSUE THREAD

**Issue by avigad** [leanprover/lean3#709]
This works:

``` lean
import data.int
open -[coercions] nat
open int
```

But this doesn't:

``` lean
import data.int
open int
open -[coercions] nat
```

The problem seems to be that "-" is interpreted as negation when int is open.

**Comment by leodemoura** [comment]
In `int/basic.lean`, you declared the token `-[`.
So, after we open `int`, the string `open -[coercions] nat` is tokenized as `open`, `-[`, `coercions`, `]`, `nat`. A possible workaround is to add a space 

``` lean
open - [coercions] nat
```

Another solution is to define the notation 

``` lean
notation `-[` n `+1]` := int.neg_succ_of_nat n    -- for pretty-printing output
```

in an auxiliary namespace (e.g. `int.extra`). Users of the `int` module will hardly ever need this notation.
If the need, they can open the auxiliary namespace.

**Comment by leodemoura** [comment]
I will close this one since this is not a bug.

**Comment by avigad** [comment]
I am sorry I didn't figure this out on my own. The notation is useful with the function definition package, to define functions on integers by cases. I implemented an easy fix by changing the notation to `-[1+n]`.

I noticed a similar conflict with notation for functions:

https://github.com/leanprover/lean/blob/master/library/algebra/function.lean#L66

But I suppose this will not cause problems, at least not with the `open` command, because it only occurs after a term has been parsed.
